matrix.js

```diff
@@ -74,7 +74,10 @@ define([], function(){
 		var isValid = _validateMatrices(a);
 
 		if(!isValid.inv)
-			throw new Error("Matrix can not be inverted as it is not a square Matrix");
+			throw {
+				type: "decomposition",
+				message: messages.get("decomposition")
```

Comment:
Should't matrix.js include messages object in this file ?

Reply:
I preferred exposing a function. I didnt want it to unnecessarily hold an object.

Reply:
This way nothing can be changed in matrix.js to show any other messages.